lighthouse-core/gather/gather-runner.js

```diff
@@ -536,7 +537,24 @@ class GatherRunner {
   static async getWebAppManifest(passContext) {
     const response = await passContext.driver.getAppManifest();
     if (!response) return null;
-    return manifestParser(response.data, response.url, passContext.url);
+    const manifest = manifestParser(response.data, response.url, passContext.url);
+
+    // Add a warning for icons that aren't fetchable.
+    if (manifest.value && Array.isArray(manifest.value.icons.value)) {
+      const iconsToFetch = manifest.value.icons.value
+        .filter(icon => icon.value.src && icon.value.src.value);
+      const fetchPromises = iconsToFetch.map(async icon => {
+        const expression = `(${pageFunctions.fetchIsOkString})(
+          ${JSON.stringify(icon.value.src.value)}
+        )`;
+        if (!await passContext.driver.evaluateAsync(expression, {useIsolation: true})) {
+          icon.warning = 'Error fetching icon';
```

Comment:
don't *love* the Promise.all mutating an object, can we make this a map of the icon results?